.github/workflows/build-macos.yml

```diff
@@ -75,36 +75,37 @@ jobs:
       fail-fast: false  # if a particular matrix build fails, don't skip the rest
       matrix:
         # Split job into 5 matrix builds, because GH actions provides 5 concurrent
-        # builds on macOS
+        # builds on macOS. This needs to be manually kept updated so that each
+        # of these builds take roughly the same time
         include:
           - { 
-            name: "x86_64 (cp36-cp310)",
+            name: "x86_64 (CPython 3.6 3.10 and above)",
             macarch: x86_64,
-            pyversions: cp36-* cp310-*
+            pyversions: "cp3{6,[1-9][0-9]}-*"
           }
 
           - { 
-            name: "x86_64 (cp37-pp37)",
+            name: "x86_64 (Python 3.7)",
             macarch: x86_64,
-            pyversions: cp37-* pp37-*
+            pyversions: "?p37-*"
           }
 
           - { 
-            name: "x86_64 (cp38-pp38)",
+            name: "x86_64 (Python 3.8)",
             macarch: x86_64,
-            pyversions: cp38-* pp38-*
+            pyversions: "?p38-*"
           }
 
           - { 
-            name: "x86_64 (cp39-pp39)",
+            name: "x86_64 (Python 3.9)",
             macarch: x86_64,
-            pyversions: cp39-* pp39-*
+            pyversions: "?p39-*"
           }
 
           - { 
-            name: "arm64 (cp38-cp39-cp310)",
+            name: "arm64 (CPython 3.8 and above)",
             macarch: arm64,
-            pyversions: cp38-* cp39-* cp310-*
+            pyversions: "cp3{8,9,[1-9][0-9]}-*"
```

Comment:
Using globs seems like it makes it less clear what exactly does what.

Reply:
Well, I think the more compact representation is better. Besides, there's a human readable name just a few lines above each of those globs anyways

Reply:
compact representation? I expected you to talk about forwards compatibility or something. Explicit is better than implicit?

Reply:
> I expected you to talk about forwards compatibility or something.

Yeah well that's another thing, but as you know in a year when we have to add 3.12 here a lot more things could change (like having to accommodate a new pypy version, or drop an older one) and we would probably have to move around stuff here a bit anyways.

Also I don't see how this is less clear, but this is subjective anyways. I see this pattern being used in many other projects using cibuildwheel (and also used in cibuildwheel docs) so I assumed people won't have much of a problem with it. Also this is not really an "implicit" thing.

Reply:
I'm not going to be a complete stick in the mud about this from the perspective of approving this PR, but this is absolutely an implicit thing.

Explicit: `cp38-* cp39-* cp310-*`
Implicit: `cp3{8,9,[1-9][0-9]}-*`

Reply:
Well, there's still a `*` at the end of what you call "explicit" so I don't really see how this is an implicit vs explicit thing
Explicit is better than implicit, but practicality beats purity :wink:

Reply:
All I'm saying is that to actually understand the new style, you have to translate it in your head into the old style, and then you can think about it.